unicode.h

```diff
@@ -24,3 +26,20 @@ int unicode_cpt_type(const std::string & utf8);
 std::string unicode_byte_to_utf8(uint8_t byte);
 uint8_t unicode_utf8_to_byte(const std::string & utf8);
 
+std::vector<std::wstring> get_gpt2_regex();
+std::vector<std::wstring> get_deepseek_coder_regex();
+std::vector<std::wstring> get_deepseek_llm_regex();
```

Comment:
I'm thinking the interface here should be:

```c++
std::vector<std::string> unicode_regex_split(const std::string & text, const std::vector<std::string> & regexes);
```

The implementation should be something like what `regex_bpe_preprocess` currently is. It loops through the regex strings and if we have a unicode representation (e.g. `"//\s?\p{L}+" -> std::wstring`) - we apply it with `std::wregex`. Else, if we have a custom implementation (for example see the GPT2 preprocess function) then we apply that.

The `unicode` model should not have any kind of notion about GPT2, Deepseek or other model-related stuff. This information should be in `llama.cpp`

Reply:
Regarding the custom implementaton, does the GPT2 preprocess function work for deepseek models? Or, we have implement the custom function for deepseek models for ourselves.